File: WEEK 2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>
#include <string.h>

int no_letter(string text);
int no_word(string text);
int no_sent(string text);
int main(void)
{
    string text = get_string("Text: ");
    int i = strlen(text);
    int letters = no_letter(text);
    int words = no_word(text);
    int sent = no_sent(text);

    float L = ((float)letters /(float)words) * 100;
    float s = ((float)sent / (float)words) * 100;
    float index = 0.0588 * L - 0.296 * s - 15.8;

    int sindex = round(index);
    if (sindex > 16)
    {
        printf("Grade 16+\n");
    }
    else if (sindex < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", sindex);
    }
}

int no_letter(string text)
{
int letter = 0;
for (int i = 0, n = strlen(text); i < n; i++)
{
    char c = text[i];
if(isalpha(c) != 0)
{
    letter++;
}
}
return letter;
}

int no_word(string text)
{
    int word = 1;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        char c = text[i];
        if (c == ' ')
        {
            word++;
        }
    }
return word;
}

int no_sent(string text)
{
    int sent = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        char c = text[i];
        if(c == '.' || c == '!' || c == '?' )
        {
            sent++;
        }
    }
    return sent;
}

File: WEEK 6/sentimental-readability/readability.py
from cs50 import get_string

n = get_string("Text:")

letter = 0
space = 0
sentence = 0

for char in n:
    if char.isalpha():
        letter += 1
    elif char ==" ":
        space += 1
    elif char == "!" or char == "." or char == "?":
        sentence += 1

words = space + 1

L = (letter/words) * 100
S = (sentence/words) * 100

grade = 0.0588 * L - 0.296 * S - 15.8

if grade > 16:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print(f"Grade {round(grade)}")


